### PR TITLE
workflows: add pr-auditor and test plans to PR templates

### PR DESCRIPTION
**Pull request by bobheadxi**
I'm aware the Helm project is actively WIP, so I'm just leaving this here for yall to merge when you feel ready :)

---

This pull request adds test plans to all PR templates, and introduces a pr-auditor workflow to ensure test plans are provided.
For more details see https://github.com/sourcegraph/sourcegraph/issues/30427 and https://docs.sourcegraph.com/dev/background-information/testing_principles

Test plan: created action should correctly set a status on this pull request.

[_Created by Sourcegraph batch change `robert/pr-auditor-rollout`._](https://k8s.sgdev.org/users/robert/batch-changes/pr-auditor-rollout)

### CODE REVIEW

File: .github/workflows/pr-auditor.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: actions/checkout@v2
+        with: { repository: 'sourcegraph/sourcegraph' }
+      - uses: actions/setup-go@v2
+        with: { go-version: '1.17' }
+
+      - run: ./dev/pr-auditor/check-pr.sh
+        env:
+          GITHUB_EVENT_PATH: ${{ env.GITHUB_EVENT_PATH }}
+          GITHUB_TOKEN: ${{ secrets.CODENOTIFY_GITHUB_TOKEN }}
+          GITHUB_RUN_URL: https://github.com/sourcegraph/infrastructure/actions/runs/${{ github.run_id }}
```
Comment (bobheadxi) [inline .github/workflows/pr-auditor.yml:19]:
```suggestion
          GITHUB_RUN_URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
```

**Review by michaellzc — APPROVED** [review]
I am down to enforce test plans for helm.

we've been adding test plans to most of our PRs lately, so we might as well enforce it for helm. The only thing I can think of is - helm is technically not GA yet, and we don't want to make any commitment. @caugustus-sourcegraph thoughts?

**Comment by bobheadxi** [comment]
Sounds good! Lets me neatly round off the batch change with all PRs merged :) Thanks yall!

**Comment by caugustus-sourcegraph** [comment]
@michaellzc - Let's run with it! I think the free-form test plan is just fine to commit to for now, we can do more formal requirements later.

**Pull request merged**